lib/engine/game/g_1849/step/bond.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+
+module Engine
+  module Game
+    module G1849
+      module Step
+        class Bond < Engine::Step::Base
+          def actions(entity)
+            return [] if !entity.corporation? || entity != current_entity
+
+            actions = []
+            actions << 'payoff_loan' if can_payoff_loan?(entity)
+            actions << 'take_loan' if @game.can_take_loan?(entity)
+            actions << 'pass' if blocks? && !actions.empty?
+
+            actions
+          end
+
+          def description
+            can_payoff_loan?(current_entity) ? 'Repay Bond' : 'Issue Bond'
+          end
+
+          def log_skip(entity)
+            super if @game.bonds? && !@game.phase.status.include?('gray_uses_white')
+          end
+
+          def take_loan_button_text
+            "Issue Bond (#{@game.format_currency(@game.loan_value)})"
+          end
+
+          def payoff_loan_button_text
+            "Repay Bond (#{@game.format_currency(@game.loan_value)})"
+          end
+
+          def can_payoff_loan?(entity)
+            !@round.issued_bond[entity] &&
+              entity.loans.any? &&
```

Comment:
Never use `.any?` unless you need to pass in a block with it. Use `!empty?`. It is significantly more efficient.